pkg/delete/internal/classic/client.go

```diff
@@ -18,34 +18,49 @@ package classic
 
 import (
 	"context"
+	"sync"
+
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/api"
 	"github.com/dynatrace/dynatrace-configuration-as-code/v2/pkg/client/dtclient"
 )
 
 func newCachedDTClient(client dtclient.Client) dtclient.Client {
 	return &cachedDTClient{
-		Client:    client,
-		listCache: make(map[urlPath]listResponse),
+		Client: client,
+		cache:  make(map[urlPath]listResponse),
 	}
 }
 
 type (
 	cachedDTClient struct {
 		dtclient.Client
-		listCache map[urlPath]listResponse
+		cache map[urlPath]listResponse
+		mu    sync.Mutex
 	}
 	urlPath      = string
 	listResponse = []dtclient.Value
 )
 
-func (client *cachedDTClient) ListConfigs(ctx context.Context, api api.API) ([]dtclient.Value, error) {
-	if _, ok := client.listCache[api.URLPath]; !ok {
-		v, err := client.Client.ListConfigs(ctx, api)
+func (c *cachedDTClient) ListConfigs(ctx context.Context, a api.API) ([]dtclient.Value, error) {
+	if _, ok := c.cache[a.URLPath]; !ok {
+		v, err := c.Client.ListConfigs(ctx, a)
 		if err != nil {
 			return nil, err
 		} else {
-			client.listCache[api.URLPath] = v
+			c.mu.Lock()
+			c.cache[a.URLPath] = v
+			c.mu.Unlock()
 		}
 	}
-	return client.listCache[api.URLPath], nil
+	return c.cache[a.URLPath], nil
+}
+
+func (c *cachedDTClient) DeleteConfigById(a api.API, id string) error {
+	// for endpoints that have a child, we
+	if a.ID == api.KeyUserActionsMobile || a.ID == api.KeyUserActionsWeb {
```

Comment:
is this needed?

Reply:
With pre dividing mechanism, it is not necessary to have that.